Research/lva-node-app/public/js/main.js

```diff
@@ -0,0 +1,638 @@
+* @param {HTMLButtonElement} htmlElement - either a string of method name, or the HTML button clicked to invoke method
+* @param {boolean} - if name is passed = true
+* @returns {Promise<any>}
+*/
+function graphEntityList(htmlElement, nameIsPassed=false) 
+{
+  return new Promise((resolve, reject) =>
+  {
+    var methodName = nameIsPassed ? htmlElement : htmlElement.getAttribute('name');
+    let payload = 
+    {
+      "@apiVersion": "1.0"
+    }
+    invokeLVAMethod(createFullPayload(methodName, payload)).then((response) =>
+    {
+      resolve(response);
+    });
+  });
+}
+
+/**
+* this payload is used for GraphInstanceDelete, GraphTopologyDelete, GraphInstanceActivate, GraphInstanceDeactivate
+* @param {string | HTMLButtonElement} htmlElement - button clicked on to invoke method on given element or string of element name (i.e. "Graph-Instance-1")
+* @param {string} method - method name
+* @param {boolean} elementNamePassed - true if function caller passed in the 
+*/
+function graphEntityModify(htmlElement, method="", elementNamePassed=false, methodNameIsPassed=false) 
+{
+  let elementName = elementNamePassed ? htmlElement : htmlElement.innerText;
+  let methodName = methodNameIsPassed ? method : htmlElement.name;
+  let payload = 
+  {
+    "@apiVersion": "1.0",
+    name: elementName
+  }
+  invokeLVAMethod(createFullPayload(methodName, payload));
+}
+
+
+/**
+* calls GraphInstanceList and GraphTopologyList. Runs on load of output.html
+* @async
+* @returns {Promise<void>}
+*/
+async function loadInstancesAndTopologies()
+{
+  getGlobals();
+  graphEntityList("GraphInstanceList", true).then(() =>
+  {
+    graphEntityList("GraphTopologyList", true);
+  });
+}
+
+/**
+* Update instance names/topology names when invoking list method on LVA
+* @param {string} methodName - Name of method invoked
+* @param {XMLHttpRequestResponseType} response - JSON response object from request
+* @returns {void}
+*/
+function updateGraphsandInstances(methodName, response) 
+{
+  let namesArray = response.payload.value;
+  if (namesArray.length != 0)
+  {
+    namesArray.forEach((item) => 
+    {
+      switch(methodName)
+      {
+        case "GraphInstanceList":
+          let instancePayload = 
+          {
+            "@apiVersion": "1.0",
+            "name": item.name,
+            "properties":
+            {
+              "topologyName": item.properties.topologyName,
+              "description": item.properties.description,
+              "parameters": item.properties.parameters
+            }
+          }
+          graphInstances[item.name] = instancePayload;
+          break;
+        
+        case "GraphTopologyList":
+          if (graphTopologies[item.name] == undefined )
+          {
+            let newGraph = new MediaGraph(item.name, item.description, item.properties.sources, item.properties.processors, item.properties.sinks, item.properties.parameters);
+            graphTopologies[item.name] = newGraph.jsonObject;
```

Comment:
i don't see the MediaGraph object used except to get it's json object. it feels like maybe it should be a method instead as a result.

Reply:
similar for graph node

Reply:
Agreed on MediaGraph. Could get rid of graph node class but i like the way the properties are formatted. Thoughts?

Reply:
I guess start with MediaGraph and I'll take a look again with the more recent changes to see how it looks with respect to GraphNode.

Reply:
Took a look. Why do you prefer the graph node be a class and not a method like MediaGraph? Seems like the exact same case to me but maybe I'm missing a detail.

Reply:
having it as a class sets us up for more functionality moving forwards, I feel. Also a class with properties is a little easier to deal with than a large json object, IMO. But I am happy to change it!

Reply:
changed to method